cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -270,30 +314,52 @@ func deviceLayoutFromPartitionTable(ptable sfdiskPartitionTable) (*DeviceLayout,
 		}
 		bd := info.BlockDevices[0]
 
+		vsType, err := fromSfdiskPartitionType(p.Type, ptable.Label)
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert sfdisk partition type %q: %v", p.Type, err)
+		}
+
 		structure[i] = gadget.VolumeStructure{
 			Name:       p.Name,
 			Size:       gadget.Size(p.Size) * sectorSize,
 			Label:      bd.Label,
-			Type:       p.Type,
+			Type:       vsType,
 			Filesystem: bd.FSType,
 		}
+
 		ds[i] = DeviceStructure{
 			LaidOutStructure: gadget.LaidOutStructure{
 				VolumeStructure: &structure[i],
 				StartOffset:     gadget.Size(p.Start) * sectorSize,
 				Index:           i + 1,
 			},
 			Node:    p.Node,
-			Created: p.isCreated(),
+			Created: isCreatedDuringInstall(&p, &bd, ptable.Label),
 		}
 	}
 
+	// if we don't have an LBA, we are probably on an MBR disk and need to
```

Comment:
why is this bit of code not using ptable.Label as condition as there rest?

Reply:
I isn't strictly necessary, `last-lba` is set only for GPT

Reply:
can you expand the TODO then:  // TODO:UC20: how/where/why/who/when to best do this ???

is not clear to me what it refers to/what we need to improve here

Reply:
Right, that TODO can be dropped now.